scripts/ci/run-integration-tests

```diff
@@ -5,49 +5,183 @@
-RETRIES=3
+CYPRESS_FOLDER="cypress"
+TESTS_FOLDER="tests"
+
+PLUGINS_FOLDER="external_plugins"
+PLUGINS_PATH=${PLUGINS_PATH:-''}
+
+
 
+## Functions
+#####################################################################
+
+# executes Cypress
+#
+# from callee:
+# FILE: file to test
+# OUTPUT_PATH: Directory where results are saved (e.g. "some/path/to/Testfile-cy.js-1")
 function executeCypress {
-  echo "=====> Executing cypress for $1"
-  local testResultPath
-  testResultPath="cypress/$(basename "$1")-result.xml"
-  npm run cypress -- -s "$1" --reporter junit --reporter-options "mochaFile=$testResultPath"
+  npm run --silent cypress -- \
+    --spec "${FILE}" \
+    --config "screenshotsFolder=${OUTPUT_PATH}/screenshots,videosFolder=${OUTPUT_PATH}/videos" \
+    --reporter junit \
+    --reporter-options "mochaFile=${OUTPUT_PATH}/result.xml"
 }
 
+# runs tests for matched file
+#
+# From callee:
+# RELATIVE_PATH: stripped down output from find command (e.g. "pages/Testfile-cy.js")
 function runTestFile {
-  echo "==> Running tests for $1"
-  i="0"
-  while [ $i -lt $RETRIES ]; do
-    i=$((i+1))
-    echo "===> Executing run $i"
-    local exit_code
-
-    # We don't want to let an error in the block below to end the entire script
-    set +e # undo instant error flag
+  echo ""
+  echo "==> Running Tests for ./${TESTS_FOLDER}/$RELATIVE_PATH"
+  local RUN=0
+  local EXIT_CODE=1
+  while [ $EXIT_CODE -ne 0 ] && [ $RUN -lt $RETRIES ]; do
+    RUN=$((RUN+1))
+    local OUTPUT_PATH="${PROJECT_ROOT}/${CYPRESS_FOLDER}/${RELATIVE_PATH}-${RUN}"
 
-    executeCypress "$1"
-    exit_code=$?
+    echo "===> Executing Cypress (${RUN}/${RETRIES})"
+    CYOUT="$(FILE="${PROJECT_ROOT}/tests/${RELATIVE_PATH}" RUN=${RUN} OUTPUT_PATH=${OUTPUT_PATH} executeCypress)"
+    EXIT_CODE=$?
 
-    set -e # redo flag
-    echo "==> $i run exit code: $exit_code"
-
-    if [ $exit_code -eq 0 ]; then
-      return 0
+    if [ $EXIT_CODE -eq 0 ]; then
+      echo "====> Successfully finished tests for ${RELATIVE_PATH}"
+    else
+      echo "====> Failures in tests for ${RELATIVE_PATH}"
+      [ $SUPPRESS_OUTPUT -eq 0 ] && echo "${CYOUT}"
     fi
   done
 
-  echo "==> Couldn't get a success in $RETRIES retries"
-  exit 1
+  if [ $EXIT_CODE -ne 0 ]; then
+    echo "===> Couldn't get a success in $RETRIES retries for $RELATIVE_PATH"
+  fi
+
+  if [ ${EXIT_CODE} -ne 0 ] && [ ${STOP_ON_FAIL} -eq 0 ]; then
+    echo "==> Continuing despite failures…"
+    return 0
+  fi
+
+  return $EXIT_CODE
+}
+
+# copies tests from plugins folder into project
+# this is necessary because Cypress doesnt allows external files to be tested…
+function setup_plugins {
+  if [ -n "${PLUGINS_PATH}" ]; then
+    echo "==> Setup external plugins…"
+    rm -rf ${PROJECT_ROOT}/${TESTS_FOLDER}/${PLUGINS_FOLDER}/
+    mkdir ${PROJECT_ROOT}/${TESTS_FOLDER}/${PLUGINS_FOLDER}/
+    rsync -aH ${PROJECT_ROOT}/${PLUGINS_PATH}/${TESTS_FOLDER}/ ${PROJECT_ROOT}/${TESTS_FOLDER}/${PLUGINS_FOLDER}/
```

Comment:
I think this doesn't work, it assumes that the tests of a plugin know where they get copied to, which seems pretty fragile to me 🤔 Either we verify / document this assumption or we need to be more hacky with the path so that it works

Reply:
The subfolder isn't the problem, see PM